Command/CreateUserCommand.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace SumoCoders\FrameworkMultiUserBundle\Command;
+
+use SumoCoders\FrameworkMultiUserBundle\User\UserRepository;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class CreateUserCommand extends ContainerAwareCommand
+{
+    protected function configure()
+    {
+        $this
+            ->setName('sumocoders:multiuser:create')
+            ->setDescription('Create a user entity')
+            ->addArgument(
+                'username',
+                InputArgument::REQUIRED,
+                'The username of the user'
+            )
+            ->addArgument(
+                'password',
+                InputArgument::REQUIRED,
+                'The password for the user'
+            )
+            ->addArgument(
+                'displayName',
+                InputArgument::REQUIRED,
+                'The display name for the user'
+            )
+            ->addOption(
+                'class',
+                null,
+                InputOption::VALUE_OPTIONAL,
+                'The class off the user',
+                'SumoCoders\FrameworkMultiUserBundle\User\User'
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $userClass = $input->getOption('class');
+
+        $repository = $this->getRepository($userClass);
+
+        $handler = new CreateUserHandler($repository);
+
+        $username = $input->getArgument('username');
+        $password = $input->getArgument('password');
+        $displayName = $input->getArgument('displayName');
+
+        $command = new CreateUser($username, $password, $displayName);
+
+        $handler->handle($command);
+
+        $output->writeln($username . ' has been created');
+    }
+
+    /**
+     * Get the repository for the user's Class.
+     *
+     * @param $userClass
+     *
+     * @return UserRepository
+     */
+    private function getRepository($userClass)
+    {
+        return $this->getContainer()->get('multi_user.user_repository.collection')->findRepositoryByClassName($userClass);
```

Comment:
Het is ook properder om geen container aware te gebruiken en je dependencies te injecteren via constructor injection.

Je zou dus `multi_user.user_repository.collection` moeten constructor injecten.